### ISSUE TITLE
Inconsistent return type of `domain`?

### ISSUE THREAD

**Issue by singularitti** [JuliaMath/Polynomials.jl#456]
I am trying two examples and using `domain` on them. But one returns a `Tuple`, the other returns a `Vector`. Could it be unified?

```julia
julia> p = Polynomial([1, 2, 3, 4])
Polynomial(1 + 2*x + 3*x^2 + 4*x^3)

julia> domain(p)
(-Inf, Inf)

julia> p = ChebyshevT([1, 0, 3, 4])
ChebyshevT(1⋅T_0(x) + 3⋅T_2(x) + 4⋅T_3(x))

julia> domain(p)
[-1, 1]
```

**Comment by jverzani** [comment]
Yes, that seems like a bug. No reason not to return a tuple in each.

**Comment by jverzani** [comment]
Oops, not a but. That is an artifact of `show`. The `domain` method returns an `Interval`.

**Comment by singularitti** [comment]
OK, I get it. Could it be `Interval(-Inf, Inf)` and `Interval[-1, 1]`?

**Comment by singularitti** [comment]
Another hint: `Intervals.jl` use `a..b` to represent intervals:

<img width="809" alt="lcfWzyIz_000009" src="https://user-images.githubusercontent.com/25192197/215365314-2e816ba9-03db-4eab-ad04-69032dc97884.png">

**Comment by jverzani** [comment]
Yeah, I didn't want to take on that dependency, but it is useful syntax. I wish `..` were in base and not a package.

**Comment by singularitti** [comment]
Maybe we could take inspirations from both. How about this?

```julia
(-Inf..Inf)
[-1..1]
```

**Comment by jverzani** [comment]
That is a show method. Would be nice, though better would be `Polynomials.Interval[-1,1]`, as it isnt' an exported thing. Want to put in a PR? Let me know, o/w I can add it.

**Comment by singularitti** [comment]
Which one do you like?

```julia
Polynomials.Interval[-1,1]
```

or `[-1..1]`?

**Comment by jverzani** [comment]
I like the latter with ( or ) for infinite values.

**Comment by singularitti** [comment]
It should work now: #460

```julia
julia> using Polynomials: domain

julia> p = Polynomial([1, 2, 3, 4])
Polynomial(1 + 2*x + 3*x^2 + 4*x^3)

julia> domain(p)
(-Inf..Inf)

julia> p = ChebyshevT([1, 0, 3, 4])
ChebyshevT(1⋅T_0(x) + 3⋅T_2(x) + 4⋅T_3(x))

julia> domain(p)
[-1..1]
```

**Comment by singularitti** [comment]
Closed by #460
